src/main/java/com/tabnine/TabNineCompletionContributor.java

```diff
@@ -172,7 +174,7 @@ TabNineProcess getProcOrPrintError() {
                 req.max_num_results = max_num_results;
                 req.region_includes_beginning = (begin == 0);
                 req.region_includes_end = (end == doc.getTextLength());
-                return proc.request(req);
+                return AppExecutorUtil.getAppExecutorService().submit(() -> proc.request(req)).get(1, TimeUnit.SECONDS);
```

Comment:
Future. get is throwing a TimeoutException if timed out,  is this handled somewhere ?

Reply:
Fuck me. That is why I want tests...

Reply:
yep, although it is not an easy thing to test...